crates/sui-indexer-alt/src/handlers/coin_balance_buckets.rs

```diff
@@ -198,8 +198,9 @@ fn get_coin_owner(object: &Object) -> Option<(StoredCoinOwnerKind, SuiAddress)>
     }
 }
 
-fn get_coin_balance_bucket(coin: &Object) -> anyhow::Result<i16> {
+pub(crate) fn get_coin_balance_bucket(coin: &Object) -> anyhow::Result<i16> {
     let Some(coin) = coin.as_coin_maybe() else {
+        // TODO: We should make this an invariant violation.
```

Comment:
Out of interest, what would the distinction between a `bail!` and an invariant violation be?

Reply:
An invariant violation would be 1) panic in debug/test mode 2) report a metric that we could set alerts on.

In this case, if this were to happen in prod, I guess this pipeline will just eventually gracefully shutdown, which would actually take a while, and we wouldn't notice until maybe when this pipeline is significantly behind?

Reply:
Errors like this will always cause the pipeline to stall, either intentionally (they cause an immediate shutdown) or because the pipeline will enter a retry loop. They should also be accompanied by an `error!` trace though, and my intention for alerting purposes is that we should alert on those. Even a single `error!` is a cause for concern and should cause a page.

Converting the error into a panic in debug mode is an interesting idea. It's helpful in pinpointing the source of the issue and failing fast, but I wonder if it makes it more likely that we would leave the DB in a bad state by stopping work half way through.  Probably it shouldn't because the indexer should be able to withstand unexpected crashes as well.